src/settings.js

```diff
@@ -856,6 +858,29 @@ var FETCH = 0; // If nonzero, enables emscripten_fetch API.
 
 var ASMFS = 0; // If set to 1, uses the multithreaded filesystem that is implemented within the asm.js module, using emscripten_fetch. Implies -s FETCH=1.
 
+var SINGLE_FILE = 0; // If set to 1, embeds all subresources in the emitted file
+                     // by converting their file names into base64 data URIs. Embedded
+                     // subresources may include (but aren't limited to) wasm, asm.js,
+                     // and static memory initialization code.
+                     //
+                     // MEM_INIT_METHOD 0 uses the same logic as SINGLE_FILE, so the following
+                     // information applies to that as well:
+                     //
+                     // If used when generating JavaScript output, the subresource data URIs
+                     // will be parsed directly in JavaScript for ensured compatibility
+                     // across runtime environments.
+                     //
+                     // If used when generating HTML output, <script> elements will be
+                     // inserted with data URI src attributes and other subresource data URIs
+                     // will be requested via XHR. This means that your Content Security Policy
+                     // may need to be updated to ensure that the script-src and/or connect-src
+                     // directives whitelist data:. If you aren't using Content Security Policy,
+                     // or your CSP header doesn't include either script-src or connect-src,
+                     // then you can safely ignore this warning.
+
 var WASM_TEXT_FILE = ''; // name of the file containing wasm text, if relevant
 var WASM_BINARY_FILE = ''; // name of the file containing wasm binary, if relevant
 var ASMJS_CODE_FILE = ''; // name of the file containing asm.js, if relevant
+
+var INCLUDE_THIRD_PARTY_SODIUMUTIL = 0; // If set to 1, sodiumutil will be included in the bundle.
```

Comment:
Is this the best way to communicate between `JS.get_subresource_location` in shared.py and jsifier.js?